go/vt/wrangler/testlib/backup_test.go

```diff
@@ -46,7 +45,31 @@ import (
 	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 )
 
+type compressionDetails struct {
+	BuiltinCompressor       string
+	BuiltinDecompressor     string
+	ExternalCompressorCmd   string
+	ExternalCompressorExt   string
+	ExternalDecompressorCmd string
+}
+
 func TestBackupRestore(t *testing.T) {
+	testBackupRestore(t, nil)
+}
+
+// TODO: @rameez. I was expecting this test to fail but it turns out
+// we infer decompressor through compression engine in builtinEngine.
+// It is only in xtrabackup where we infer decompressor through extension & BuiltinDecompressor param.
```

Comment:
Can you point to the code where this is happening?
There should be no inconsistency between builtin engine and xtrabackup engine

Reply:
In built-in engine logic for decompression is 
	if compress {
		var decompressor io.ReadCloser

		if *ExternalDecompressorCmd != "" {
			decompressor, err = newExternalDecompressor(ctx, *ExternalDecompressorCmd, reader, params.Logger)
		} else {
			decompressor, err = newBuiltinDecompressor(compressionEngine, reader, params.Logger)
		}
--> see newBuiltInDecompressor take compressionEngine as argument instead of decompressionEngine. It might be a bug?

========================================================================================
Whereas in XtraEngine logic is you infer if it is "auto" then get it from extension else use whatever is provided as decompression value.

if compressed {
			var decompressor io.ReadCloser

			if *ExternalDecompressorCmd != "" {
				decompressor, err = newExternalDecompressor(ctx, *ExternalDecompressorCmd, reader, logger)
			} else {
				decompressor, err = newBuiltinDecompressorFromExtension(extension, *BuiltinDecompressor, reader, logger)
			}

Reply:
@rvrangel , can you confirm this ?

Reply:
This is because in the case of the builtin engine,`compressionEngine` comes from the manifest. It was added here:
https://github.com/vitessio/vitess/blob/13f5e071a1f818181fd13a168d97efb416e5c519/go/vt/mysqlctl/builtinbackupengine.go#L74-L79

and gets passed to `restoreFile` by `restoreFiles` here:
https://github.com/vitessio/vitess/blob/13f5e071a1f818181fd13a168d97efb416e5c519/go/vt/mysqlctl/builtinbackupengine.go#L609

Since we don't have an extension in the builtin engine like we have on `xtrabackup`, we can't infer it, so we rely on the `CompressionEngine` reported on the manifest file when the backup was first taken.

Reply:
I realise now this may be counter intuitive as the `-builtin_decompressor` would effectively be ignored in this case. we could probably add an extra check in the `builtin` engine to use the `CompressionEngine` from the manifest if it is `auto`, or just override it with what as passed via `-builtin_decompressor`, that would probably be more consistent with the `xtrabackup` engine, right?

Reply:
so basically you are saying .. 
line#609 in builtinbackupengine will change to (roughly like below)

var decomEngine = bm.CompressionEngine
if *builtinDecompression != "auto"{
 decompEngine = *builtinDecompression
} 
err := be.restoreFile(ctx, params, bh, &fes[i], bm.TransformHook, !bm.SkipCompress, decompEngine, name)

Reply:
I like this .. its more intuitive..

Reply:
yeah, that's exactly what I was thinking!